src/coreclr/src/jit/lower.cpp

```diff
@@ -3449,25 +3450,26 @@ void Lowering::LowerCallStruct(GenTreeCall* call)
         return;
     }
 
-#if defined(FEATURE_HFA)
-    if (comp->IsHfa(call))
+    if (GlobalJitOptions::compFeatureHfa)
     {
+        if (comp->IsHfa(call))
+        {
 #if defined(TARGET_ARM64)
-        assert(comp->GetHfaCount(call) == 1);
+            assert(comp->GetHfaCount(call) == 1);
 #elif defined(TARGET_ARM)
-        // ARM returns double in 2 float registers, but
-        // `call->HasMultiRegRetVal()` count double registers.
-        assert(comp->GetHfaCount(call) <= 2);
-#elif  // !TARGET_ARM64 && !TARGET_ARM
-        unreached();
+            // ARM returns double in 2 float registers, but
+            // `call->HasMultiRegRetVal()` count double registers.
+            assert(comp->GetHfaCount(call) <= 2);
+#else  // !TARGET_ARM64 && !TARGET_ARM
+            NYI("Unknown architecture");
```

Comment:
why change this from `unreached()` to `NYI`?

Reply:
It turns out that part of the #if statement was never reached because of the #if which used to wrap it, and unreached() causes a compiler error now as the followon statements start producing errors. I'll change it to NOWAY from NYI per your other comment.

Reply:
Upon reflection, this should actually probably remain as an NYI.

---
In reply to: [511061420](https://github.com/dotnet/runtime/pull/43706#discussion_r511061420) [](ancestors = 511061420)